packages/replay/src/session/getSession.ts

```diff
@@ -36,8 +36,8 @@ export function getSession({
 
     if (!isExpired) {
       return { type: 'saved', session };
-    } else if (session.sampled === 'error') {
-      // Error samples should not be re-created when expired, but instead we stop when the replay is done
+    } else if (session.sampled === 'buffer') {
+      // Buffered samples should not be re-created when expired, but instead we stop when the replay is done
```

Comment:
We need to change this -  see https://github.com/getsentry/sentry-javascript/issues/7929

I verified this in a local app, and the unexpected thing here is that this also means if you have an ongoing error (or buffer) sampled session, then this expires, and a new user action happens, it will _always_ be unsampled afterwards, even if you have error sample rate 1. Instead, what we _want_ is to only stop sampling if a buffered session was started to be captured due to an error or `capture()`. This probably needs a bit refactoring for this to work...!

Reply:
Makes sense -- we could write to a new property on the session when this occurs and then check for that here instead of checking for `sampled`?

Reply:
I made a PR to add this to this PR: https://github.com/getsentry/sentry-javascript/pull/7931